crates/agentgateway/src/proxy/httpproxy.rs

```diff
@@ -532,6 +538,11 @@ impl HTTPProxy {
 		let selected_backend =
 			select_backend(selected_route.as_ref(), &req).ok_or(ProxyError::NoValidBackends)?;
 		let selected_backend = resolve_backend(selected_backend, self.inputs.as_ref())?;
+		// Set backend info as soon as we have resolved to add the metric labels
```

Comment:
whats the motivation for this? It seems reasonable but then do we need it again on L873? Seems moving it makes sense but not clear why duplicating it?

Reply:
I was getting "unknown" for backend in the metric label here without this change:
```
agentgateway_response_bytes_total{backend="unknown",protocol="http",method="GET",status="200",reason="Upstream",bind="8080/default/http-gw-for-test",gateway="default/http-gw-for-test",listener="http",route="default/httpbin",route_rule="unknown"} 1090
```

Maybe there's another spot it needs to get set, but setting it here seemed to work.